velox/exec/TableScan.cpp

```diff
@@ -117,6 +118,9 @@ RowVectorPtr TableScan::getOutput() {
       if (!split.hasConnectorSplit()) {
         noMoreSplits_ = true;
         dynamicFilters_.clear();
+        if (splitTracer_ != nullptr) {
```

Comment:
Let's determine this based on operator type? And noMoreInput won't be called on a source operator? Thanks!

Reply:
>And noMoreInput won't be called on a source operator? 

Yes, so that we can not call `finishTrace()` in operator::noMoreInput.
```C++
void addInput(RowVectorPtr /* unused */) override {
    VELOX_FAIL("SourceOperator does not support addInput()");
  }

  void noMoreInput() override {
    VELOX_FAIL("SourceOperator does not support noMoreInput()");
  }
```

Reply:
Let's just call this in Operator::finishTrace()? Let the framework handle the tracing calls? Thanks!